### PR TITLE
[wasm] Optimize output from threaded test runner

### PR DESCRIPTION
**Pull request by maraf**
- Pass verbosity from xharness server to the browser
- Use logger instead of Console to respect verbosity
- Drop custom message process from xharness server
- Log test output as Debug
- Align test ouput for threaded and threadless runners

Contributes to https://github.com/dotnet/runtime/issues/88763

<details>
  <summary>Threadless - ok</summary>

```
info: Application arguments: --run WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll -method System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success -notrait category=OuterLoop -notrait category=failing -verbosity Info
info: [main 17:01:43.503] MONO_WASM: starting script http://127.0.0.1:63389/_framework/dotnet.js
info: [main 17:01:43.504] MONO_WASM: starting in http://127.0.0.1:63389/_framework/
info: Initializing dotnet version 9.0.0-dev commit hash f1583e0558e353d8de0f2816c9d7aa1c89fbe999
info: Discovering: System.Net.WebSockets.Client.Tests.dll (method display = ClassAndMethod, method display options = None)
info: Discovered:  System.Net.WebSockets.Client.Tests.dll (found 1 of 247 test cases)
info: Starting:    System.Net.WebSockets.Client.Tests.dll
warn: Blocking on the main thread is very dangerous, see https://emscripten.org/docs/porting/pthreads.html#blocking-on-the-main-browser-thread
info: Finished:    System.Net.WebSockets.Client.Tests.dll
info: Received expected 1462 of C:\Development\dotnet\runtime\artifacts\bin\System.Net.WebSockets.Client.Tests\Debug\net9.0-browser\browser-wasm\AppBundle\xharness-output\testResults.xml
info: Finished writing 1462 bytes of RESULTXML
info: Xml file was written to the provided writer.
info: Tests run: 2 Passed: 2 Inconclusive: 0 Failed: 0 Ignored: 0 Skipped: 0
info: test-main.js exiting WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll with result 0
info: [0x000f3b0c 17:01:46.373] MONO_WASM: forceDisposeProxies done: 6 imports, 0 exports, 0 GCHandles, 0 JSHandles.
info: WASM EXIT 0
info: Waiting to flush log messages with a timeout of 120 secs ..
XHarness exit code: 0
```

</details>
<details>
  <summary>Threadless - error</summary>

```
info: Incoming arguments: --setenv=DOTNET_TEST_WEBSOCKETHOST=127.0.0.1:59788 --setenv=DOTNET_TEST_HTTPHOST=127.0.0.1:59788 --setenv=DOTNET_TEST_REMOTE_LOOP_HOST=127.0.0.1:59788 --setenv=DOTNET_TEST_SECUREWEBSOCKETHOST=127.0.0.1:59789 --setenv=DOTNET_TEST_SECUREHTTPHOST=127.0.0
.1:59789 --setenv=DOTNET_TEST_HTTP2HOST=127.0.0.1:59789 --setenv=XHARNESS_LOG_TEST_START=true --no-memory-snapshot --no-memory-snapshot --setenv=IsBrowserThreadingSupported=true --run WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll -method System.Net.WebSockets.Clien
t.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success -notrait category=OuterLoop -notrait category=failing -verbosity Info
info: Application arguments: --run WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll -method System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success -notrait category=OuterLoop -notrait category=failing -verbosity Info
info: [main 12:39:18.296] MONO_WASM: starting script http://127.0.0.1:59788/_framework/dotnet.js
info: [main 12:39:18.297] MONO_WASM: starting in http://127.0.0.1:59788/_framework/
info: Initializing dotnet version 9.0.0-dev commit hash f1583e0558e353d8de0f2816c9d7aa1c89fbe999
info: Discovering: System.Net.WebSockets.Client.Tests.dll (method display = ClassAndMethod, method display options = None)
info: Discovered:  System.Net.WebSockets.Client.Tests.dll (found 1 of 247 test cases)
info: Starting:    System.Net.WebSockets.Client.Tests.dll
info: [STRT] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: ws://127.0.0.1:59788/WebSocket/EchoWebSocket.ashx)
info: [FAIL] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: ws://127.0.0.1:59788/WebSocket/EchoWebSocket.ashx)
info: Assert.True() Failure
info: Expected: True
info: Actual:   False
info:    at System.Net.WebSockets.Client.Tests.SendReceiveTest.SendReceive_Concurrent_Success(Uri server) in C:\Development\dotnet\runtime\src\libraries\System.Net.WebSockets.Client\tests\SendReceiveTest.cs:line 398
info: --- End of stack trace from previous location ---
info: [STRT] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: wss://127.0.0.1:59789/WebSocket/EchoWebSocket.ashx)
info: [FAIL] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: wss://127.0.0.1:59789/WebSocket/EchoWebSocket.ashx)
info: Assert.True() Failure
info: Expected: True
info: Actual:   False
info:    at System.Net.WebSockets.Client.Tests.SendReceiveTest.SendReceive_Concurrent_Success(Uri server) in C:\Development\dotnet\runtime\src\libraries\System.Net.WebSockets.Client\tests\SendReceiveTest.cs:line 398
info: --- End of stack trace from previous location ---
info: Finished:    System.Net.WebSockets.Client.Tests.dll
info: Received expected 2178 of C:\Development\dotnet\runtime\artifacts\bin\System.Net.WebSockets.Client.Tests\Debug\net9.0-browser\browser-wasm\AppBundle\xharness-output\testResults.xml
info: Finished writing 2178 bytes of RESULTXML
info: Xml file was written to the provided writer.
info: Tests run: 2 Passed: 0 Inconclusive: 0 Failed: 2 Ignored: 0 Skipped: 0
info: test-main.js exiting WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll with result 1
info: WASM EXIT 1
info: Waiting to flush log messages with a timeout of 120 secs ..
fail: Application has finished with exit code TESTS_FAILED but 0 was expected
XHarness exit code: 71 (GENERAL_FAILURE)
```

</details>
<details>
  <summary>Threaded - ok</summary>

```
info: Application arguments: --run WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll -method System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success -threads -notrait category=OuterLoop -notrait category=failing -verbosity Info
info: [main 17:00:23.642] MONO_WASM: starting script http://127.0.0.1:63324/_framework/dotnet.js
info: [main 17:00:23.642] MONO_WASM: starting in http://127.0.0.1:63324/_framework/
info: Initializing dotnet version 9.0.0-dev commit hash f1583e0558e353d8de0f2816c9d7aa1c89fbe999
info: Configured filters:
info:   XUnitFilter [Type: Single; include; Method: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success]
info:   XUnitFilter [Type: Trait; exclude; Trait name: category; Trait value: OuterLoop]
info:   XUnitFilter [Type: Trait; exclude; Trait name: category; Trait value: failing]
info:   XUnitFilter [Type: Single; include; Method: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success]
info: Assembly: System.Net.WebSockets.Client.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (System.Net.WebSockets.Client.Tests.dll)
warn: Blocking on the main thread is very dangerous, see https://emscripten.org/docs/porting/pthreads.html#blocking-on-the-main-browser-thread
info: [Test environment: 32-bit .NET .NET 9.0 [collection-per-class, non-parallel]]
info: [Test framework: xUnit.net 2.6.7.0]
info:
info: Test collection for System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest
info: [PASS] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success
info: [PASS] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success
info: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest 0.1845299 ms
info: Received expected 1462 of C:\Development\dotnet\runtime\artifacts\bin\System.Net.WebSockets.Client.Tests\Debug\net9.0-browser\browser-wasm\AppBundle\xharness-output\testResults.xml
info: Finished writing 1462 bytes of RESULTXML
info: Xml file was written to the provided writer.
info: Tests run: 248 Passed: 2 Inconclusive: 0 Failed: 0 Ignored: 246 Skipped: 0
info: test-main.js exiting WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll with result 0
info: [0x000f3b0c 17:00:26.707] MONO_WASM: forceDisposeProxies done: 6 imports, 0 exports, 0 GCHandles, 0 JSHandles.
info: WASM EXIT 0
info: Waiting to flush log messages with a timeout of 120 secs ..
XHarness exit code: 0
```

</details>
<details>
  <summary>Threaded - error</summary>

``` 
info: Application arguments: --run WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll -method System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success -threads -notrait category=OuterLoop -notrait category=failing -verbosity Info
info: [main 13:43:40.505] MONO_WASM: starting script http://127.0.0.1:61321/_framework/dotnet.js
info: [main 13:43:40.505] MONO_WASM: starting in http://127.0.0.1:61321/_framework/
info: Initializing dotnet version 9.0.0-dev commit hash f1583e0558e353d8de0f2816c9d7aa1c89fbe999
info: Configured filters:
info:   XUnitFilter [Type: Single; include; Method: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success]
info:   XUnitFilter [Type: Trait; exclude; Trait name: category; Trait value: OuterLoop]
info:   XUnitFilter [Type: Trait; exclude; Trait name: category; Trait value: failing]
info:   XUnitFilter [Type: Single; include; Method: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success]
info: Assembly: System.Net.WebSockets.Client.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (System.Net.WebSockets.Client.Tests.dll)
warn: Blocking on the main thread is very dangerous, see https://emscripten.org/docs/porting/pthreads.html#blocking-on-the-main-browser-thread
info: [Test environment: 32-bit .NET .NET 9.0 [collection-per-class, non-parallel]]
info: [Test framework: xUnit.net 2.6.7.0]
info:
info: Test collection for System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest
info: [STRT] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: ws://127.0.0.1:61321/WebSocket/EchoWebSocket.ashx)
info: [FAIL] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: ws://127.0.0.1:61321/WebSocket/EchoWebSocket.ashx)
info: Assert.True() Failure
info: Expected: True
info: Actual:   False
info:    at System.Net.WebSockets.Client.Tests.SendReceiveTest.SendReceive_Concurrent_Success(Uri server) in C:\Development\dotnet\runtime\src\libraries\System.Net.WebSockets.Client\tests\SendReceiveTest.cs:line 418
info: --- End of stack trace from previous location ---
info: [STRT] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: wss://127.0.0.1:61322/WebSocket/EchoWebSocket.ashx)
info: [FAIL] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: wss://127.0.0.1:61322/WebSocket/EchoWebSocket.ashx)
info: Assert.True() Failure
info: Expected: True
info: Actual:   False
info:    at System.Net.WebSockets.Client.Tests.SendReceiveTest.SendReceive_Concurrent_Success(Uri server) in C:\Development\dotnet\runtime\src\libraries\System.Net.WebSockets.Client\tests\SendReceiveTest.cs:line 418
info: --- End of stack trace from previous location ---
info: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest 0.1870299 ms
info: Received expected 2428 of C:\Development\dotnet\runtime\artifacts\bin\System.Net.WebSockets.Client.Tests\Debug\net9.0-browser\browser-wasm\AppBundle\xharness-output\testResults.xml
info: Finished writing 2428 bytes of RESULTXML
info: Xml file was written to the provided writer.
info: Tests run: 248 Passed: 0 Inconclusive: 0 Failed: 2 Ignored: 246 Skipped: 0
info: test-main.js exiting WasmTestRunner.dll System.Net.WebSockets.Client.Tests.dll with result 1
info: WASM EXIT 1
info: Waiting to flush log messages with a timeout of 120 secs ..
fail: Application has finished with exit code TESTS_FAILED but 0 was expected
```

</details>

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs [hunk 1]
```diff
@@ -275,9 +276,32 @@ private string BuildUrl(ServerURLs serverURLs)
                 sb.Append('&');
 
             sb.Append($"arg={HttpUtility.UrlEncode(arg)}");
-        }
+        }
+
+        if (sb.Length > 0)
+            sb.Append('&');
+
+        sb.Append($"arg=-verbosity&arg={HttpUtility.UrlEncode($"{VerbosityToString()}")}");
```
Comment (pavelsavara) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs:284]: `HttpUtility.UrlEncode` is overkill for simple enum with known values.

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs [hunk 2]
```diff
@@ -19,5 +19,7 @@ public static void WriteOnSingleLine(XElement assembliesElement)
             Console.WriteLine($"STARTRESULTXML {bytes.Count} {base64} ENDRESULTXML");
             Console.WriteLine($"Finished writing {bytes.Count} bytes of RESULTXML");
         }
+
+        public static string EscapeNewLines(string message) => message.Replace("\r", "\\r").Replace("\n", "\\n");
```
Comment (pavelsavara) [inline src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs:23]: we also have some data driven unit tests with nasty unicode codepoints. 
It makes test names, which the python downstream can't parse. 
Ending up with `�` when things are happy or losing the log if not.
I wonder if you want to also `String.Normalize` or something.

**Review by pavelsavara — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs [hunk 3]
```diff
@@ -275,9 +276,32 @@ private string BuildUrl(ServerURLs serverURLs)
...
         return uriBuilder.ToString();
-    }
+    }
+
+    //MinimumLogLevel.Critical,
+    //MinimumLogLevel.Error,
+    //MinimumLogLevel.Warning,
+    //MinimumLogLevel.Info,
+    //MinimumLogLevel.Debug,
+    //MinimumLogLevel.Verbose
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs:295]: nit: space after `//` and sort the lines to be the same order as the switch below

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs (hunk 2 above)
Comment (akoeplinger) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs:23]: does this happen with the standard xunit test runner too? I wonder if we can copy some normalization code from there.

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmThreadedTestRunner.cs [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
 internal class WasmThreadedTestRunner : XUnitTestRunner
 {
     public WasmThreadedTestRunner(LogWriter logger) : base(logger)
     {
+        TestStagePrefix = string.Empty;
+    }
+
+    protected override string ResultsFileName { get => string.Empty; set => throw new InvalidOperationException("This runner outputs its results to stdout."); }
+
+    protected override void HandleTestFailed(ITestFailed msg)
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmThreadedTestRunner.cs:23]: why is just this method overridden?

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs [hunk 5]
```diff
@@ -93,9 +93,6 @@ public override async Task Run(IEnumerable<TestAssemblyInfo> testAssemblies)
             _assembliesElement.Add(resultsXmlAssembly);
         }
         TotalTests = totalSummary.Total;
-        Console.WriteLine($"{Environment.NewLine}=== TEST EXECUTION SUMMARY ==={Environment.NewLine}Total: {totalSummary.Total}, Errors: 0, Failed: {totalSummary.Failed}, Skipped: {totalSummary.Skipped}, Time: {TimeSpan.FromSeconds((double)totalSummary.Time).TotalSeconds}s{Environment.NewLine}");
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs:96]: I'd keep this, it's quite nice to have a summary at the end when looking at console logs.

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs [hunk 6]
```diff
@@ -64,21 +64,21 @@ public override async Task Run(IEnumerable<TestAssemblyInfo> testAssemblies)
 
             if (EnvironmentVariables.IsLogTestStart())
             {
-                testSink.Execution.TestStartingEvent += args => { Console.WriteLine($"[STRT] {args.Message.Test.DisplayName}"); };
+                testSink.Execution.TestStartingEvent += args => { OnInfo($"[STRT] {args.Message.Test.DisplayName}"); };
             }
             testSink.Execution.TestPassedEvent += args =>
             {
-                Console.WriteLine($"[PASS] {EscapeNewLines(args.Message.Test.DisplayName)}");
+                OnDebug($"[PASS] {WasmXmlResultWriter.EscapeNewLines(args.Message.Test.DisplayName)}");
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs:71]: WasmXmlResultWriter doesn't feel the right place for EscapeNewLines given how it is used.
Probably better to move it to https://github.com/dotnet/xharness/blob/main/src/Microsoft.DotNet.XHarness.Common/Utilities/StringUtils.cs

**Review by akoeplinger — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmThreadedTestRunner.cs (hunk 4 above)
Comment (maraf) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmThreadedTestRunner.cs:23]: The original implementation does a pretty extensive dump. We are logging less details in single threaded

```
info: [FAIL] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success
info: [FAIL] System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success
info:    Test name: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: ws://127.0.0.1:60267/WebSocket/EchoWebSocket.ashx)
info:    Test case: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success
info:    Assembly:  [System.Net.WebSockets.Client.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
info:    Exception messages: Assert.True() Failure
info: Expected: True
info: Actual:   False
info:    Exception stack traces:    at System.Net.WebSockets.Client.Tests.SendReceiveTest.SendReceive_Concurrent_Success(Uri server) in C:\Development\dotnet\runtime\src\libraries\System.Net.WebSockets.Client\tests\SendReceiveTest.cs:line 418
info: --- End of stack trace from previous location ---
info:    Execution time: 0.1452448
info:  **** Output start ****
info: GetConnectedWebSocket: ConnectAsync starting.
info: GetConnectedWebSocket: ConnectAsync done.
info:
info:  **** Output end ****
info:
info:    Test name: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success(server: wss://127.0.0.1:60268/WebSocket/EchoWebSocket.ashx)
info:    Test case: System.Net.WebSockets.Client.Tests.ArraySegmentSendReceiveTest.SendReceive_Concurrent_Success
info:    Exception messages: Assert.True() Failure
info: Expected: True
info: Actual:   False
info:    Exception stack traces:    at System.Net.WebSockets.Client.Tests.SendReceiveTest.SendReceive_Concurrent_Success(Uri server) in C:\Development\dotnet\runtime\src\libraries\System.Net.WebSockets.Client\tests\SendReceiveTest.cs:line 418
info: --- End of stack trace from previous location ---
info:    Execution time: 0.0540651
info:  **** Output start ****
info: GetConnectedWebSocket: ConnectAsync starting.
info: GetConnectedWebSocket: ConnectAsync done.
info:
info:  **** Output end ****
```

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs (hunk 5 above)
Comment (maraf) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs:96]: There is summary implemented in the inherited class. I have removed a duplicate, though the one from the threadless was more eye catching 🤷‍♂️

```
info: Tests run: 2 Passed: 2 Inconclusive: 0 Failed: 0 Ignored: 0 Skipped: 0
```

Comment (pavelsavara) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs:96]: Add the `=== TEST EXECUTION SUMMARY ===` to the common code ?

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs (hunk 2 above)
Comment (maraf) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs:23]: Looking at xunit source, I have found this smaller beast https://github.com/xunit/xunit/blob/2210888189e19227f4b0413f55146792f69bdf84/src/xunit.v3.runner.common/Reporters/DefaultRunnerReporterMessageHandler.cs#L98 and this bigger one https://github.com/xunit/xunit/blob/2210888189e19227f4b0413f55146792f69bdf84/src/xunit.v3.runner.common/Sinks/DelegatingSinks/DelegatingXmlCreationSink.cs#L400

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs (hunk 6 above)
Comment (maraf) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs]: The `Microsoft.DotNet.XHarness.Common` is referenced by CLI, but not by runner projects. Anyway I have added a new StringUtils class in `Microsoft.DotNet.XHarness.TestRunners.Common`

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmThreadedTestRunner.cs (hunk 4 above)
Comment (akoeplinger) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmThreadedTestRunner.cs:25]: I'd be inclined to fix this in the base class, we could log the additional output only in the Debug log case

File: src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs (hunk 2 above)
Comment (maraf) [reply src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs]: IIRC the line replacement was there to ensure that `WasmTestMessagesProcessor` (the server-side message pump) was seeing PASS/FAIL messages as a single line and thus as a single message and thus it was changing the scope (from info to debug) correctly. I have removed this scope changing logic in this PR, so I should be safe to remove the line replacement 🤞

File: src/Microsoft.DotNet.XHarness.TestRunners.Common/TestRunner.cs [hunk 7]
```diff
@@ -137,11 +139,13 @@ protected void OnAssemblyFinish(Assembly asm)
 
     protected void LogFailureSummary()
     {
-        if (FailureInfos == null || FailureInfos.Count == 0)
+        if (!ShowFailureInfos || FailureInfos == null || FailureInfos.Count == 0)
         {
             return;
         }
 
+        Console.WriteLine($"MF LogFailureSummary {FailureInfos.Count}");
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.XHarness.TestRunners.Common/TestRunner.cs:147]: leftover from debugging?

**Review by akoeplinger — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.TestRunners.Common/TestRunner.cs (hunk 7 above)
Comment (maraf) [reply src/Microsoft.DotNet.XHarness.TestRunners.Common/TestRunner.cs:147]: Thanks!
